### DISCUSSION TITLE
Crawled property remapping does not appear in filters

### DISCUSSION THREAD

**Discussion by blue2cat** [microsoft-search/pnp-modern-search#2953] [Q&A] [answered]
I am attempting to create search filters for a search box using several custom columns that have been created for my SharePoint site. The process of mapping crawled properties to their respective `RefinableString` managed properties has thus far been unsuccessful. The steps I have followed are as follows:
1. Map crawled properties to RefinableStrings. 
![image](https://user-images.githubusercontent.com/46635587/236366138-f3e854f2-2f3f-411d-b076-519e6628d6a1.png)

2. Reindex sites. 
3. Added `RefinableString`  0 --> 3 in the `Selected Properties` section. 
![image](https://user-images.githubusercontent.com/46635587/236366386-cbd65120-27a7-4f47-bc82-e82b94ba0936.png)
 4. Added `RefinableString` 0 --> 3 to search filters.

Now, what happens is `RefinableString00` works great. There are no issues filtering by `SiteTitle`. The rest of the mapped properties are not functioning as expected. `RefinableString03` has a number of values, but the total is less than expected. The values do, however, show up in the filters web part. `RefinableString01` and `RefinableString02` do not show up at all. When I filter the  `Query text` in the `Search Results` section for the missing refinable strings, I get several values, but well below the number I was expecting. This may be more of a SharePoint issue, but I wanted to raise it here to see if anyone has troubleshooting ideas.

**Comment by kasperbolarsen** [comment, marked as answer]
Hi @blue2cat , as always the best approach is to verify the data from the API first in order to rule out that it might be a pnp-modern-search issue.
I suggest checking the data using either the SharePoint Search Query tool or the SP Editor ( a Chrome extension)

**Reply by afnepipe** [reply]
@kasperbolarsen could you add more information/details about the data checking?

**Reply by kasperbolarsen** [reply]
Sure. Knowing for sure which value a given property has, as well as knowing which properties are available on an object gives us a solid foundation for figuring out why our data values might be wrong or missing. 
I have made a short series of YouTube videos about the tools available,  and you can find the links here, https://ms365thinking.blogspot.com/2023/04/new-videos-on-youtube-useful-tools-when.html

**Reply by blue2cat** [reply]
@kasperbolarsen, thank you for your helpful documentation. Using the SP Editor, I have determined that the issue is in fact a upstream datatype issue. Thank you for pointing me to the SP Editor, it proved to be a useful tool in sleuthing out the real cause of the issue.

**Reply by afnepipe** [reply]
@kasperbolarsen
I have the same issue. Some refinable ones are not shown in the result and not in the filters either. I have been trying with the extension, but the search option is disabled. I tried with PnP search tool and the results are the same as in PnP Search. 
What else can I do to see the results in the missing columns? Is this error related to the ReIndex or Crawling?
![image](https://github.com/microsoft-search/pnp-modern-search/assets/96144314/ab207e29-c567-4347-99f7-c27115233672)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/96144314/195009c0-fcc3-4360-aeb5-b92163d41994)

**Reply by kasperbolarsen** [reply]
Hi @afnepipe 
Once you have mapped the crawled properties to your Refinablestings/ints/bools I would recommend that you go into site settings in a site collection where you use those Site Columns and set the site to be reindex (in Search and Offline availability). That *should* ensure that the mapping will take hold within 1-2 hours, otherwise you should expect that it will take up to 24 hours before the RefinableStrings gets values.

**Reply by afnepipe** [reply]
Thanks for your answer @kasperbolarsen 
All mapping configuration, Site ReIndex and Offline ReIndex was made on May 5th. As you can see, only some columns started to show values but not all of them.
I created a support case at Microsoft to validate if the issue is just in the ReIndexing.
Any other recommendation?

**Reply by kasperbolarsen** [reply]
You could check when one or more of the items was indexed using https://pnp.github.io/powershell/cmdlets/Get-PnPSearchCrawlLog.html
